content/client-server-api/modules/instant_messaging.md

```diff
@@ -385,6 +386,49 @@ An example of a media message with a caption is:
 Clients MUST render the caption alongside the media and SHOULD prefer its
 formatted representation.
 
+##### Mathematical messages
+
+{{% added-in v="1.11" %}}
+
+Users might want to send mathematical notations in their messages.
+
+To send mathematical notations clients MUST use the `formatted_body` and
+therefore the `org.matrix.custom.html` format, described above. This makes
+mathematical notations valid on any `msgtype` which can support this format
+appropriately.
+
+Mathematical notations themselves use the `span` or `div` tags, depending
+whether the notation should be presented inline or not. The mathematical
+notation is written in [LaTeX](https://www.latex-project.org/) format using the
+`data-mx-maths` attribute.
+
+The contents of the tag should be a fallback representation for clients that
+cannot render the LaTeX format. The fallback representation could be, for
+example, an image, or an HTML approximation, or the raw LaTeX source. When using
+an image as a fallback, the sending client should be aware of issues that may
+arise from the receiving client using a different background colour. The `body`
+should include a textual representation of the notation.
+
+An example of a mathematical notation is:
+
+```json
+{
+  "msgtype": "m.text",
+  "format": "org.matrix.custom.html",
+  "body": "This is an equation: sin(x)=a/b.",
+  "formatted_body": "This is an equation:
+      <span data-mx-maths=\"\\sin(x)=\\frac{a}{b}\">
+        sin(<i>x</i>)=<sup><i>a</i></sup>/<sub><i>b</i></sub>
+      </span>"
+}
+```
+
+The LaTeX format is poorly defined and has several extensions, so if a client
+encounters syntax that it cannot render, it SHOULD present the fallback
+representation instead. Clients SHOULD, however, aim to support, at minimum, the
+basic LaTeX2e maths commands and the TeX maths commands, with the possible
+exception of commands that could be security risks.
```

Comment:
I think we need to call out the security risk more here, maybe by putting [this paragraph](https://github.com/matrix-org/matrix-spec-proposals/pull/2191/files#diff-9f3ff21f46a094d6bbf0f87639fa549aa118651ce79ac01bbae8668bfa5c4f70R119-R125) from the MSC in a WARNING box.

Reply:
I agree, I added the warning box.

Reply:
I think it might also be worth calling out that calling out to a full-fledged LaTeX compiler such as `xelatex` is not recommended. Instead, using libraries which support only a subset of the spec should be expected.

Reply:
I made the warning more comprehensive to include that.